bundles/framework/publisher2/tools/AbstractPluginTool.js

```diff
@@ -107,8 +107,10 @@ function(sandbox, mapmodule, localization, instance, handlers) {
                 me.stop();
             }
         }
-        var event = Oskari.eventBuilder('Publisher2.ToolEnabledChangedEvent')(me);
-        sandbox.notifyAll(event);
+        if (!Oskari.util.isMobile()) {
```

Comment:
This is the wrong place to solve the issue. Things start to go really weird really quickly at some point if "tool was enabled/disabled" events are sent or NOT sent depending on the map size.